src/test/java/org/shikshalokam/backend/scp/TestScpPermissionCRUDOperations.java

```diff
@@ -39,117 +39,139 @@ public void init() {
     public void testCreatePermissionWithValidPayload() {
         logger.info("Started calling the CreatePermission API:");
         try {
-            RestAssured.baseURI = PropertyLoader.PROP_LIST.get("scp.qa.api.base.url").toString();
             createPermissionEndpoint = new URI(PROP_LIST.get("scp.create.permission.endpoint").toString());
         } catch (URISyntaxException e) {
             throw new RuntimeException(e);
         }
 
-        // Updated request body
-        String requestBody = "{ " +
-                "\"code\": \"project_test_break\", " +
-                "\"module\": \"permissions\", " +
-                "\"request_type\": [\"POST\"], " +
-                "\"api_path\": \"/scp/v1/projects/create\", " +
-                "\"status\": \"ACTIVE\" " +
-                "}";
+        // Create JSON object dynamically using JSONObject
+        JSONObject requestBody = new JSONObject();
+        String prefix = "project_"; // Define your desired prefix
+        requestBody.put("code", generateRandomWord(prefix, 10));
+        requestBody.put("module", "permissions");
+        JSONArray requestTypeArray = new JSONArray();
+        requestTypeArray.add("POST");
+        requestBody.put("request_type", requestTypeArray);
+        requestBody.put("api_path", "/scp/v1/projects/create");
+        requestBody.put("status", "ACTIVE");
 
         // Use the extracted internalAccessToken in the request header
         Response response = given()
-                .header("X-auth-token", "bearer " + internalAccessToken)  // Using the extracted token
+                .header("X-auth-token", "bearer " + internalAccessToken)
                 .contentType(ContentType.JSON)
-                .body(requestBody)
+                .body(requestBody.toString())
                 .when().post(createPermissionEndpoint);
 
         // Log the status code and response body
         int statusCode = response.getStatusCode();
         String responseBody = response.getBody().asString();
         logger.info("Response Status Code: " + statusCode);
         logger.info("Response Body: " + responseBody);
+
         // Pretty print the response to the console or logs
         response.prettyPrint();
 
-        //log the end call for createPermission API
+        // Validate response code is 201
+        Assert.assertEquals(statusCode, 201, "Status code should be 200");
+
+        // Fetch ID from response and validate
+        int createdId = response.jsonPath().getInt("result.id");
+        Assert.assertNotNull(createdId, "Created ID should not be null");
+        logger.info("Created ID: " + createdId);
+
+        // Log the end call for createPermission API
         logger.info("Ended calling the CreatePermission API with valid payload.");
+    }
 
+    //Method to generate random word as code(Project name)
```

Comment:
RandomStringUtils.randomAlphabetic(20) use this method .Bulit in methods more optimized